burn-tensor/src/tensor/activation/base.rs

```diff
@@ -69,3 +69,8 @@ pub fn log_sigmoid<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B
 pub fn silu<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B, D> {
     tensor.clone().mul(sigmoid(tensor))
 }
+
+/// Applies the tanh function
+pub fn tanh<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B, D> {
+    tensor.tanh()
+}
```

Comment:
Since this is brand new activation function, we will need a unit test similar to silu: https://github.com/antimora/burn/blob/828ef56cb381b57f0e6581117ba39cff101223a0/burn-tensor/src/tests/activation/silu.rs#L2 

To add a test: 
1. add a file under tests/activation
2. make sure to put a macro with a correct name: #[burn_tensor_testgen::testgen(tanh)] 
3. add to mod.rs like this: https://github.com/antimora/burn/blob/828ef56cb381b57f0e6581117ba39cff101223a0/burn-tensor/src/tests/activation/mod.rs#L4
4. add a module call like this: https://github.com/antimora/burn/blob/828ef56cb381b57f0e6581117ba39cff101223a0/burn-tensor/src/tests/mod.rs#L15


To test the activation function, run `cargo test` under `burn-ndarray` (or other backends). Tensor tests run with a backend.

Reply:
Really thank you for the guides. Will complete this part later.

Reply:
Oh I just noticed that there's already a unit test for `tanh` but under `ops` dir. Should we keep this? Or just move it under `activation`, for they actually have the same test data and logic.

Reply:
I don't think the test is necessary, since it is already tested as a tensor operation. However adding this alias can be more intuitive.

Reply:
I think we can keep the new activation test to stay consistent with other activation funcs. It's indeed more intuitive for new maintainers to find calling chain and understand the project struct. And these test snippet are not huge burdens for `tensor` code base. Though the long name and redundancy are really ugly. 😅

Reply:
I would leave it. For any public API, we should always have tests. It just happens that the underlying code/function is already tested but there is no guaranty unless we have tests. Also have tests will ensure confidence if we need to do a refactor.

Reply:
Ok it should exist there. We can keep both.